File: 2016/j2.py
# code by savir singh

row1 = input().split(" ")
row2 = input().split(" ")
row3 = input().split(" ")
row4 = input().split(" ")
total = 0

r1 = []
r2 = []
r3 = []
r4 = []
c1 = []
c2 = []
c3 = []
c4 = []

for i in range(4):
    r1.append(int(row1[i]))
    r2.append(int(row2[i]))
    r3.append(int(row3[i]))
    r4.append(int(row4[i]))

c1.append(r1[0])
c1.append(r2[0])
c1.append(r3[0])
c1.append(r4[0])
c2.append(r1[1])
c2.append(r2[1])
c2.append(r3[1])
c2.append(r4[1])
c3.append(r1[2])
c3.append(r2[2])
c3.append(r3[2])
c3.append(r4[2])
c4.append(r1[3])
c4.append(r2[3])
c4.append(r3[3])
c4.append(r4[3])

total = r1[0] + r1[1] + r1[2] + r1[3]

if r2[0] + r2[1] + r2[2] + r2[3] == total and r3[0] + r3[1] + r3[2] + r3[3] == total and r4[0] + r4[1] + r4[2] + r4[3] == total and c1[0] + c1[1] + c1[2] + c1[3] == total and c2[0] + c2[1] + c2[2] + c2[3] == total and c3[0] + c3[1] + c3[2] + c3[3] == total and c4[0] + c4[1] + c4[2] + c4[3] == total:
    print("magic")
else:
    print("not magic")

File: 2016/j2.cpp
// code by savir singh

#include <bits/stdc++.h>
using namespace std;
// macros
#define int long long
#define double long double
#define bit32 int32_t
#define del erase
#define pb push_back
#define str string
#define scani(x) scanf("%lld", &x)
#define scan(x) scanf("%s", &x)
#define scanc(x) scanf("%c", &x)
#define printi(x) printf("%lld", x);
#define print(x) printf("%s", x);
#define printc(x) printf("%c", x);

// driver code
bit32 main()
{
    int num1, num2, num3, num4, num5, num6, num7, num8, num9, num10, num11, num12, num13, num14, num15, num16;
    cin >> num1 >> num2 >> num3 >> num4 >> num5 >> num6 >> num7 >> num8 >> num9 >> num10 >> num11 >> num12 >> num13 >> num14 >> num15 >> num16;
    int total=num1+num2+num3+num4;
    if (num5+num6+num7+num8!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    if (num9+num10+num11+num12!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    if (num13+num14+num15+num16!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    if (num1+num5+num9+num13!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    if (num2+num6+num10+num14!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    if (num3+num7+num11+num15!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    if (num4+num8+num12+num16!=total) {
        cout << "not magic" << "\n";
        return false;
    }
    cout << "magic" << "\n";
}
